### ISSUE TITLE
Paru keeps updating package without reinstalling it after every reboot

### ISSUE THREAD

**Issue by flakusha** [Morganamilo/paru#763]
### Affected Version
paru v1.10.0 - libalpm v13.0.1

### Description
I have checked previous issues.
It appears that paru keeps updating package, even if it is already last version because of it's versioning format.
Pamac does not have this issue.

This package is:
[proton-ge-custom](https://aur.archlinux.org/packages/proton-ge-custom)

### Output
```
:: Synchronizing package databases...
 core is up to date
 extra is up to date
 community is up to date
 multilib is up to date
 realtime is up to date
:: Starting full system upgrade...
 there is nothing to do
:: Looking for AUR upgrades
:: Looking for devel upgrades
:: Resolving dependencies...
:: Calculating conflicts...
:: Calculating inner conflicts...
:: packages not in the AUR: mangohud  mangohud-common
:: marked out of date: amdgpu-pro-libgl  feroxbuster  unreal-engine
:: orphans: python2-cairo  python2-numpy

Aur (1) proton-ge-custom-2:GE.Proton7.18-1
```

<details><summary>pacman.conf and paru.conf</summary>

```sh
#
# /etc/pacman.conf
#
# See the pacman.conf(5) manpage for option and repository directives

#
# GENERAL OPTIONS
#
[options]
# The following paths are commented out with their default values listed.
# If you wish to use different paths, uncomment and update the paths.
#RootDir     = /
#DBPath      = /var/lib/pacman/
CacheDir = /var/cache/pacman/pkg/
#LogFile     = /var/log/pacman.log
#GPGDir      = /etc/pacman.d/gnupg/
#HookDir     = /etc/pacman.d/hooks/
HoldPkg      = pacman glibc manjaro-system
# If upgrades are available for these packages they will be asked for first
SyncFirst    = manjaro-system archlinux-keyring manjaro-keyring
#XferCommand = /usr/bin/curl -L -C - -f -o %o %u
#XferCommand = /usr/bin/wget --passive-ftp -c -O %o %u
#CleanMethod = KeepInstalled
#UseDelta    = 0.7
Architecture = auto

#IgnorePkg   =
#IgnorePkg   =
#IgnoreGroup =

#NoUpgrade   =
#NoExtract   =

# Misc options
#UseSyslog
#Color
#NoProgressBar
# We cannot check disk space from within a chroot environment
CheckSpace
#VerbosePkgLists
ParallelDownloads = 10

# By default, pacman accepts packages signed by keys that its local keyring
# trusts (see pacman-key and its man page), as well as unsigned packages.
SigLevel    = Required DatabaseOptional
LocalFileSigLevel = Optional
#RemoteFileSigLevel = Required

# NOTE: You must run `pacman-key --init` before first using pacman; the local
# keyring can then be populated with the keys of all official Manjaro Linux
# packagers with `pacman-key --populate archlinux manjaro`.

#
# REPOSITORIES
#   - can be defined here or included from another file
#   - pacman will search repositories in the order defined here
#   - local/custom mirrors can be added here or in separate files
#   - repositories listed first will take precedence when packages
#     have identical names, regardless of version number
#   - URLs will have $repo replaced by the name of the current repo
#   - URLs will have $arch replaced by the name of the architecture
#
# Repository entries are of the format:
#       [repo-name]
#       Server = ServerName
#       Include = IncludePath
#
# The header [repo-name] is crucial - it must be present and
# uncommented to enable the repo.
#

# The testing repositories are disabled by default. To enable, uncomment the
# repo name header and Include lines. You can add preferred servers immediately
# after the header, and they will be used before the default mirrors.

[core]
SigLevel = PackageRequired
Include = /etc/pacman.d/mirrorlist

[extra]
SigLevel = PackageRequired
Include = /etc/pacman.d/mirrorlist

[community]
SigLevel = PackageRequired
Include = /etc/pacman.d/mirrorlist

# If you want to run 32 bit applications on your x86_64 system,
# enable the multilib repositories as required here.

[multilib]
SigLevel = PackageRequired
Include = /etc/pacman.d/mirrorlist

# An example of a custom package repository.  See the pacman manpage for
# tips on creating your own repositories.
#[custom]
#SigLevel = Optional TrustAll
#Server = file:///home/custompkgs

[realtime]
Server = https://pkgbuild.com/~dvzrv/repos/realtime/$arch

# $PARU_CONF
# /etc/paru.conf
# ~/.config/paru/paru.conf
#
# See the paru.conf(5) manpage for options

#
# GENERAL OPTIONS
#
[options]
PgpFetch
Devel
Provides
DevelSuffixes = -git -cvs -svn -bzr -darcs -always -hg
#AurOnly
#BottomUp
#RemoveMake
#SudoLoop
#UseAsk
#SaveChanges
#CombinedUpgrade
#CleanAfter
#UpgradeMenu
#NewsOnUpgrade

#LocalRepo
#Chroot = ~/hdd/tmp/pacman/
#Sign
#SignDb
#KeepRepoCache
CloneDir = /home/flak/hdd/tmp/pacman/

#
# Binary OPTIONS
#
#[bin]
#FileManager = vifm
#MFlags = --skippgpcheck
#Sudo = doas

```

</details>

**Comment by Morganamilo** [comment]
The package pulls from git sources and is therefore a -git package.

**Comment by flakusha** [comment]
Oh, I see.
If there is any possibility to check and fix this kind of behavior I can look into it.

**Comment by Morganamilo** [comment]
The packager should not use git versions of things or pin the versions.

**Comment by flakusha** [comment]
Okay, let's leave it this way then :smile:

**Comment by loathingKernel** [comment]
I am the packager of that package and I respectfully disagree. The package in question gets a specific commit from the upstream repository and uses `git submodule` to get commits of the other git repositories. This is not something I invented nor copied from an unreliable resource, I took it from other similar packages that exist in the Arch repositories, such as https://github.com/archlinux/svntogit-community/blob/packages/ppsspp/trunk/PKGBUILD.
